extensions/blocks/story/edit.native.js

```diff
@@ -91,201 +92,215 @@ class StoryEdit extends React.Component {
 			payload.newId,
 			payload.mediaUrl
 		);
-		setAttributes({ mediaFiles: updatedMediaFiles });
-		this.setState({ isSaveInProgress: false });
+		setAttributes( { mediaFiles: updatedMediaFiles } );
+		this.setState( { isSaveInProgress: false } );
+	}
+
+	onStoryPressed() {
+		const { attributes } = this.props;
+
+		if ( this.state.isUploadInProgress ) {
+			// issue cancellation for all media files involved
+			requestMediaFilesUploadCancelDialog( attributes.mediaFiles );
+		} else if ( this.state.isSaveInProgress ) {
+			requestMediaFilesSaveCancelDialog( attributes.mediaFiles );
+		} else if ( this.state.didUploadFail ) {
+			requestMediaFilesFailedRetryDialog( attributes.mediaFiles );
+		} else {
+			// open the editor
+			this.onEditButtonTapped();
+		}
 	}
 
 	render() {
 		const { attributes, isSelected } = this.props;
 		const { mediaFiles } = attributes;
-		const hasContent = !!mediaFiles.length;
+		const hasContent = !! mediaFiles.length;
 		const { isUploadInProgress, isSaveInProgress, didUploadFail, didSaveFail } = this.state;
 
 		return (
-			<View style={styles['wp-story-container']}>
-				{!hasContent && (
-					<Text style={styles['wp-story-wrapper']}>Empty Story placeholder here</Text>
-				)}
-				{hasContent && (
-					<View style={{ flex: 1 }}>
-						{!isUploadInProgress && !isSaveInProgress && isSelected && (
-							<StoryEditingButton onEditButtonTapped={this.onEditButtonTapped} />
-						)}
-						<BlockMediaUpdateProgress
-							coverUrl={mediaFiles[0].url} // just select the first one // TODO see how to handle video
-							mediaFiles={mediaFiles}
-							onUpdateMediaUploadProgress={this.updateMediaUploadProgress}
-							onFinishMediaUploadWithSuccess={this.finishMediaUploadWithSuccess}
-							onFinishMediaUploadWithFailure={this.finishMediaUploadWithFailure}
-							onMediaUploadStateReset={this.mediaUploadStateReset}
-							onUpdateMediaSaveProgress={this.updateMediaSaveProgress}
-							onFinishMediaSaveWithSuccess={this.finishMediaSaveWithSuccess}
-							onFinishMediaSaveWithFailure={this.finishMediaSaveWithFailure}
-							onMediaSaveStateReset={this.mediaSaveStateReset}
-							onFinalSaveResult={this.onStorySaveResult}
-							onMediaModelCreated={this.onMediaModelCreated}
-							renderContent={({
-								isUploadInProgress,
-								isUploadFailed,
-								isSaveInProgress,
-								isSaveFailed,
-								retryMessage,
-							}) => {
-								return (
-									<Image
-										url={mediaFiles[0].url} // just select the first one // TODO see how to handle video
-										style={styles['wp-story-image']}
-									/>
-								);
-							}}
-						/>
-					</View>
-				)}
-			</View>
+			<TouchableWithoutFeedback
+				accessible={ ! isSelected }
+				onPress={ this.onStoryPressed }
+				disabled={ ! isSelected }
+			>
+				<View style={ styles[ 'wp-story-container' ] }>
+					{ ! hasContent && (
+						<Text style={ styles[ 'wp-story-wrapper' ] }>
+							This Story is empty. Tap to add media.
```

Comment:
I'd guess we should localise this, right? 🤔

Reply:
you're right - this is changed in this PR https://github.com/Automattic/jetpack/pull/17539 👍